resource_management/azure_mgmt_storage/spec/spec_helper.rb

```diff
@@ -14,32 +12,37 @@
 include Azure::ARM::Resources
 include Azure::ARM::Storage
 
-def create_resource_group
-  resource_group_name = get_random_name('RubySDKTest_')
-  params = Azure::ARM::Resources::Models::ResourceGroup.new
-  params.location = 'westus'
+class ResourceHelper
+  attr_accessor :storage_client
+  attr_accessor :resource_client
 
-  RESOURCES_CLIENT.resource_groups.create_or_update(resource_group_name, params).value!.body
-end
+  def initialize
+    tenant_id = ENV['AZURE_TENANT_ID']
+    client_id = ENV['AZURE_CLIENT_ID']
+    secret = ENV['AZURE_CLIENT_SECRET']
+    subscription_id = ENV['AZURE_SUBSCRIPTION_ID']
 
-def delete_resource_group(name)
-  RESOURCES_CLIENT.resource_groups.delete(name).value!
-end
+    token_provider = ApplicationTokenProvider.new(tenant_id, client_id, secret)
+    credentials = TokenCredentials.new(token_provider)
 
-def get_random_name(prefix, length = 1000)
-  prefix + SecureRandom.uuid.downcase.delete('^a-zA-Z0-9')[0...length - prefix.length]
-end
+    @storage_client = StorageManagementClient.new(credentials)
+    @storage_client.long_running_operation_retry_timeout = ENV['RETRY_TIMEOUT'].to_i || 30
```

Comment:
I added it to streamline developer and test experience independent. This means when we are running the tests with `INTEG_RECORDED ` we'll retry at every` ENV['RETRY_TIMEOUT'`] (whcih is set to 0) otherwise running live test take default to 30 secs. 

This way we speed up the recorded tests with `INTEG_RECORDED ` environment variable but not in live one.